indico/modules/events/models/principals.py

```diff
@@ -27,7 +27,12 @@ class EventPrincipal(PrincipalPermissionsMixin, db.Model):
 
     @declared_attr
     def __table_args__(cls):
-        return auto_table_args(cls, schema='events')
+        group_perm_constraint = db.CheckConstraint(
+            '(type != {} AND type != {}) OR (NOT CAST(permissions AS text[]) && ARRAY[\'paper_editing\','
```

Comment:
`type NOT IN ({}, {})` and I think `CAST(permission AS text[])` can be simplified to `permissions::text[]`.